File: recommender-server/src/models/course.py
from sqlmodel import Field, SQLModel
from datetime import time

class Course(SQLModel, table=True):
    """
    Representation of a course in the database
    """
    id: int | None = Field(default=None, primary_key=True)
    title: str = Field()
    day: int = Field()
    type: int = Field()
    site: int = Field()
    code: str = Field()
    start: time = Field()
    end: time = Field()
    track: int = Field()
    semester: int = Field()
    description: str = Field()
    url: str = Field()


class CourseCreate(SQLModel):
    title: str
    day: int
    type: int
    site: int
    code: str
    start: time = Field()
    end: time = Field()
    track: int
    semester: int
    description: str
    url: str


class CourseRead(SQLModel):
    id: int
    title: str
    day: int
    type: int
    site: int
    code: str
    start: time
    end: time
    track: int
    semester: int
    description: str
    url: str

File: recommender-vue/src/models/course.ts
export interface Course extends Record<string, any> {
  id: number
  title: string
  day: number
  type: number
  site: number
  code: string
  start: string
  end: string
  track: number
  semester: number
  description: string
  url: string
}

export const calculateDuration = (course: Course) => {
  const startVal = course.start.split(':').map((str) => Number.parseInt(str))

  const endVal = course.end.split(':').map((str) => Number.parseInt(str))

  const start = new Date()
  const startMili = start.setHours(startVal[0], startVal[1], 0, 0)

  const end = new Date()
  const endMili = end.setHours(endVal[0], endVal[1], 0, 0)

  return (endMili - startMili) / 60000
}

export const dayToString = (course: Course) => {
  const days = ['On Appointment', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday']

  return days[course.day]
}
